src/plugins/discourse/weights.js

```diff
@@ -87,3 +100,22 @@ export function _trustLevelWeight(trustLevel: number | null): NodeWeight {
   }
   return weight;
 }
+
+export function _categoryWeight(
+  category: MirrorCategoryId,
+  {categoryWeights, defaultCategoryWeight}: WeightsConfig
+): NodeWeight {
+  const weight = categoryWeights.get(category);
+  if (weight != null) return weight;
+
+  return defaultCategoryWeight;
+}
+
+export function _weightFromTags(
+  tags: $ReadOnlyArray<Tag>,
+  {tagWeights, defaultTagWeight}: WeightsConfig
+): NodeWeight {
+  const weightForTag = (tagId: TagId) =>
+    either(tagWeights.get(tagId), defaultTagWeight);
+  return tags.reduce((acc, tag) => acc * weightForTag(tag), 1);
```

Comment:
i cant reply to the old review, but re discussion [here](https://github.com/sourcecred/sourcecred/pull/2644#discussion_r557017642) from @wchargin:

yeah, i see how this is actually the logical semantics for defaultTagWeight -- good point.